### PR TITLE
use OSSLzu instead of lu format for size_t display

### PR DESCRIPTION
**Pull request by edelangh**
##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: apps/passwd.c [hunk 1]
```diff
@@ -556,7 +556,7 @@ static char *shacrypt(const char *passwd, const char *magic, const char *salt)
     OPENSSL_strlcat(out_buf, "$", sizeof out_buf);
     if (rounds_custom) {
         char tmp_buf[7 + 9 + 1]; /* "rounds=999999999" */
-        sprintf(tmp_buf, "rounds=%lu", rounds);
+        sprintf(tmp_buf, "rounds=%zu", rounds);
```
Comment (mattcaswell) [inline apps/passwd.c]: Not all platforms have support for `zu`. Please use the `OSSLzu` macro instead defined in e_os2.h

**Comment by edelangh** [comment]
Ok, I juste change zu to OSSLzu, thank for information

**Comment by edelangh** [comment]
Ok I juste add CLA: trivial to my commit

**Comment by mattcaswell** [comment]
"Need CLA" label removed. Needs a second team review.

**Comment by levitte** [comment]
I assume this is for master and 1.1.0

**Comment by richsalz** [comment]
The format string is wrong, it needs to be
```
        sprintf(tmp_buf, "rounds=%"OSSLzu, rounds);
```

Percent is missing.

**Comment by richsalz** [comment]
ping @edelangh please make the fix and post an amended commit.

File: apps/passwd.c [hunk 2]
```diff
@@ -556,7 +556,7 @@ static char *shacrypt(const char *passwd, const char *magic, const char *salt)
     OPENSSL_strlcat(out_buf, "$", sizeof out_buf);
     if (rounds_custom) {
         char tmp_buf[7 + 9 + 1]; /* "rounds=999999999" */
```
Comment (bernd-edlinger) [inline apps/passwd.c]: Aehm, the buffer is too short for 64bit integers. (9 is even too short for 32bit integers)

**Comment by richsalz** [comment]
took of the second-review tag; this needs some changes, please.

**Comment by edelangh** [comment]
Sorry for taking so much time, I put it on my TODO LATER list, and I forgetted

File: apps/passwd.c (hunk 2 above)
Comment (richsalz) [inline apps/passwd.c]: yes, this should probably be "```[sizeof("rounds=999999999"]``` or similar.

Comment (bernd-edlinger) [inline apps/passwd.c]: max. unsigned long is 4294967295, 10 digits if long=32 bit
and 18446744073709551615, thus 20 digits if long=64 bit.

Comment (richsalz) [inline apps/passwd.c]: yeah, just be safe and make it an 80char buffer.  seriously.

**Comment by edelangh** [comment]
Now I think now it's will be good

File: apps/passwd.c [hunk 3]
```diff
@@ -555,8 +555,8 @@ static char *shacrypt(const char *passwd, const char *magic, const char *salt)
     OPENSSL_strlcat(out_buf, magic, sizeof out_buf);
     OPENSSL_strlcat(out_buf, "$", sizeof out_buf);
     if (rounds_custom) {
-        char tmp_buf[7 + 9 + 1]; /* "rounds=999999999" */
-        sprintf(tmp_buf, "rounds=%lu", rounds);
+        char tmp_buf[80]; /* "rounds=999999999" */
```
Comment (levitte) [inline apps/passwd.c]: That's some future proofing! ;-)

**Comment by richsalz** [comment]
It's new code in master only.  and now this is committed, thanks. dbaa069

**Pull request closed without merging**